### PR TITLE
Proposal for Contributor Experience SIG

### PR DESCRIPTION
**Pull request by prophen**
@ExitoLab and I are requesting approval to co-found a Contributor Experience SIG for the Spinnaker project. 

The Contributor Experience Special Interest Group (SIG) would be responsible for improving the experience of those who contribute upstream to the Spinnaker project. We do this by creating, and maintaining programs and processes that promote community health and reduce project friction while retiring those programs and processes that don’t. Being conscientious of our contributor base is critical to scaling the project, growing the ecosystem, and helping the project succeed.

Note to reviewers: It might make sense to squash the commits. There are lots of small updates as it was a collaborative effort.

### CODE REVIEW

**Comment by ExitoLab** [comment]
@prophen Thank you for the invitation and for the big move.. Yes, the PR looks good

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:15]: According to this language, it sounds like you're implicitly looking for administrative access to these tools. Is that correct? Would like clarification if not. If so, how do you plan to manage access control to these systems?

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:22]: I'm interested in more detail, but I don't see how the organization homepage would be providing any more clarity / details?

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:23]: Items in this list seem to have direct, conflicting overlaps with the [Summit SIG](https://github.com/spinnaker/governance/tree/master/sig-summit). Is this SIG meant to replace the Summit SIG? Are the leads of the Summit SIG aware of this?

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:31]:
```suggestion
- Mentoring programs, such as: office hours, meet the contributors, 1:1 pairing, group mentoring, etc.
```

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:27]:
```suggestion
  - Other events, like other SIG face-to-face events, upon request and consideration
```

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:28]: FYI: https://github.com/spinnaker/governance/pull/51

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:33]: 1. What are the planned contents of this repository? 
1. Who will need write access to it?

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:57]: 72 business hours is 9 business days. Is this language intentional?

Regardless, I suspect using "business days" is going to lead to problems: Business days are locale-specific, which the Spinnaker organization is not (Canada does not have the same business holidays as the United States, for example).

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:65]: What about code for the in-scope projects and services? I'm not clear on whether or not this SIG will be writing code in support of its initiatives, or looking for other groups to perform the legwork. For example, there's _some_ code around contributorship today, are you planning to take over these scripts?

In my mind, the SIG should be responsible for the scripts and tooling that comes out of this SIG if it applies to the project as a whole, including scripts that may be a part of the governance repo itself (e.g. https://github.com/spinnaker/governance/tree/master/scripts and more).

If this SIG is _not_ responsible for the code and tooling that will support the processes and policies this SIG sets out to create, I have little optimism in the short- or long-term success of this SIG's overall goals.

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:68]: I'm unclear on how this point co-exists with "user community management" being out of scope.

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:73]:
```suggestion
This SIG adheres to the Roles and Organization Management outlined in Spinnaker's governance policy and will actively suggest updates to those policies as part of our work.
```

File: cx-sig-charter-draft.md
Comment (robzienert) [inline cx-sig-charter-draft.md:78]:
```suggestion
Chairs SHOULD plan Contributor Experience meetings monthly, at the least, and should strive to enable annual face-to-face interaction as conditions allow. Chairs will also ensure that technical changes from subprojects follow the process change communication guidelines we establish.
```

File: cx-sig-proposed-README.md
Comment (robzienert) [inline cx-sig-proposed-README.md:20]: Clarity needed: coordinate communication of what?

**Review by robzienert — COMMENTED** [review]
Love the idea, but I'm unclear on the practical scope of this SIG. 

This SIG at present sounds like it is a policy and process generator, but one that outsources the implementation of these policies and processes to other SIGs. If this is the case, I'm concerned about the actual effectiveness this SIG will have. We (the TOC) have had a lot of difficulty enacting policies and processes that require SIGs to perform actions, or need to write code in support of this work. 

In my mind, this SIG would be responsible for implementing and maintaining any supporting tooling and code that is needed for automating these systems where needed.

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:23]: To my knowledge this is not a replacement

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:15]: @robzienert some aspects of this access will be managed through automation that the contributor experience will build. This building phase will create groups to handle things. For example, slack automation. That will not be blanket access but rather controlled by a process that will be based on GitOps, much like we have in the Kubernetes community.

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:28]: Yes #51 is a good issue. What this meant by mentorship here is 1:1 and group mentoring. The idea is a new user comes into the community and test the waters and when they are ready a mentor will guide them within a SIG. Again, very much like we do in Jenkins and Kubernetes.

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:33]: Content of this repository will be automation specific to this SIG (e.g, guides, various automation specific to contribex)
I feel to start the access will just be for leads of the contribEx sig.

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:57]: We may need to revisit this time box

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:65]: The contribEx SIG will be writing the code for the initiatives specific to contribex. For example, if there is slack automation, various automation around owners for this SIG/repo, that is where this will live and be handled by the contribex team.

File: cx-sig-charter-draft.md
Comment (markyjackson-taulia) [reply cx-sig-charter-draft.md:68]: We can help with the formation of SIG's and/or working groups but the idea initiation is up to the users wishing to form so said group. That is different than community management.

File: cx-sig-proposed-README.md
Comment (markyjackson-taulia) [reply cx-sig-proposed-README.md:20]: 1 idea I see here is the frequency by which a SIG updates the larger community. Think of a community meeting where SIG are chosen to give an update to the community, that is where this will happen/

**Comment by markyjackson-taulia** [comment]
@robzienert I would love to meet and walk you through some of the ideas here with @prophen & @ExitoLab

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:23]: An important distinction here is **contributor** events, which to refers to a contributor day or hackathon event specifically for a project contributor audience. I think that the Spinnaker Summit, which historically has been a conference event, is out of scope. That event is going to be managed by the Linux Foundation events team going forward, in collaboration with the Summit SIG, which will help develop the speaker content program and make platform/program decisions with the LF team in the same way other CDF events are managed.  This SIG can cooperate with the Summit SIG as necessary to add contributor events to the Summit as desired.

IMO: Producing events is a lot of work. Breaking it up this way, by audience, will help us use open processes to to the work collaboratively across stakeholder organizations when planning events. This is LF methodology and prevents one company from dominating the decisions.

@michaelgalloway and @gcomstock, please let us know your thoughts as the co-leads of the SIG : )

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:15]: Good question @robzienert. Public platform access control is technically managed by the Continuous Delivery Foundation. My observation is that an additional human interface with the CDF is needed in addition to what the CDF provides to facilitate sharing and make sure that access info is communicated to fairly empower contributors while. CDF can help with secure sharing.

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:22]: This is a good catch. We need to add more clarification and reference prior art from other projects as needed (which is how this oversight came about in the first place, lol).

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:65]: @robzienert, can you please link to the existing code around contributorship to for the sake of this conversation? I'd love to request that existing artifacts be referenced specifically in this PR.

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:68]: I see how this wording is unclear and will think about a better suggestion. I think it is trying to say that general user community management is out of scope, as in, this SIG is not wholly or primarily responsible for user community management (which is in scope for the Steering Committee in our docs as written, cc @imosquera). However, this group can assist in ensuring that SIGs to address projects needs are created, and that critical/productive SIGs are maintained as people move in and out of the project. They can do this through enablement and relationship management, which intersects with, and is ideally positioned by, other work described in the charter.

File: cx-sig-proposed-README.md
Comment (dnilasor) [reply cx-sig-proposed-README.md:20]: Adding clarification here is a good idea, and nice example from @markyjackson-taulia

To me this is a pretty general designation that a SIG Lead's responsibility, among other things, is to coordinate communication with SIG members and community stakeholders in the SIG's interest area about the SIG's activities (meetings, initiatives, decisions, etc.)

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:31]: Detail but, it's important to note that this group would be launching a mentorship program (so maybe scope should say that clearly). And then, maintaining it. Perhaps that meaning is the intent of the word "new" and could be clarified further.

Prior art on mentorship program, FWIW: https://github.com/spinnaker/governance/pull/51

File: cx-sig-charter-draft.md
Comment (robzienert) [reply cx-sig-charter-draft.md:65]: I linked to the code in my original comment.

**Comment by robzienert** [comment]
Had a chat with @markyjackson-taulia, which brought some needed clarity, namely that this SIG will be focusing on itself and will be operating largely in isolation as far as policies and processes go, offering then to help other SIGs onboard _if they want to opt-in_. It's an implicit goal that the improvements made by this SIG will be alluring enough that the fragmentation of community processes will decrease over the medium- and long-term as other SIGs fall inline. My question of who does the work is that the SIG will be doing its own work, which is 💯 .

Furthermore, this SIG is aiming to help with holistic community communication - injecting themselves periodically into other SIGs to aggregate and disseminate status updates and other communication to the broader community. I'm a personal fan of this, as it's been a self-identified problem by the SIG Leads, but there hasn't been any group to take on the responsibility of solving these problems.

I do think this proposal needs some messaging cleanup, specifically around its strategy: It was not clear to me at all how this SIG interacts with other SIGs. All-in-all, I'm pretty excited by the premise of this SIG and what it could mean for improving contributorship and overall community health.

+1 given some tightening around messaging.

File: cx-sig-charter-draft.md
Comment (gcomstock) [reply cx-sig-charter-draft.md:23]: Hi @dnilasor,

I don't have much to add here. Producing events is a lot of work for sure, and my involvement with this year's summit was thin. I'm all in support. Thank you!

**Review by ajordens — APPROVED** [review]
I'd like to see this move forward.

This is an ambitious plan, but a needed effort. I appreciate that there's a 6-month check-in to see how things are going.

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:65]: Is that the only such existing asset that you would propose this group should maintain, or are there other assets?

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:22]: @prophen @ExitoLab this is an easy fix, please check the referenced doc in the K8S project (and consult other porjects if you want), then you can just define whatever you want to say here instead of referencing another link. You can delete this bullet entirely if you want but I think that executing some GitHub mgmt tasks as described might be in scope - your call.

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:57]: I think this time box should be revisted after the SIG is founded and determines what the capacity is for response. I recommend deleting this and adding that determination as a goal @ExitoLab @Prophen

File: cx-sig-charter-draft.md
Comment (robzienert) [reply cx-sig-charter-draft.md:65]: I think that's it. All membership management has been a manual affair thus far. Other github actions we are maintaining are covered by other SIGs.

File: cx-sig-charter-draft.md
Comment (ezimanyi) [inline cx-sig-charter-draft.md:52]: Does this mean that this SIG is empowered to establish policies about how all SIGs conduct their business?  (For example, requiring a particular platform for meetings, requiring a particular meeting cadence, etc.).

Similar to the above question, I'm wondering how much oversight/approval we'd expect the TOC/SC to provide in these cases.

File: cx-sig-charter-draft.md
Comment (ezimanyi) [inline cx-sig-charter-draft.md:15]: I think some examples of the types of policies that might be implemented would be useful.  Depending on the particular policy, I would imagine that we might want to have this SIG work on building consensus and making a proposal, then presenting that to the SC/TOC for comments/final approval.

I don't want to add extra steps for small tactical procedures, but do want to make sure that we are appropriately reviewing changes that have a broader impact (and thus why understanding some of the types of policies would be useful).

**Review by ezimanyi — COMMENTED** [review]
This looks good to me.  I agree that it is ambitious, but I think these would all be great improvements to the community, and the SIG can prioritize appropriately to make progress on these things.

I think my biggest question here is around what policies/procedure adopted by the SIG would need to get approval from the SC/TOC before being implemented, and left two specific questions to that effect.  This is a pretty broad charter that includes setting policies that affect all contributors to the community, and want to understand the degree to which the SIG would be empowered to do this alone vs. draft proposals for the TOC/SC.

As noted on the specific comments, specific examples might be helpful for discussion, as I don't want to add a lot of steps that slow progress but also want to make sure there is appropriate oversight of changes that will broadly impact the community.

File: cx-sig-proposed-README.md
Comment (prophen) [reply cx-sig-proposed-README.md:20]:
```suggestion
* Coordinate communication with SIG participants and the larger community.
```

File: cx-sig-charter-draft.md
Comment (dnilasor) [reply cx-sig-charter-draft.md:15]: Nikema is working on adding some more specifics to the doc in response to your suggestion. One thing I'll say generally is, the model we discussed with Rob in his comments is more of an "opt-in" model than an approval gate model. This SIG can establish tools/resources/processes to enable SIGs to manage things or make them more welcoming/equitable. Then next steps would be to pitch or offer those things to the SIGs themselves. If SIGs choose to adopt them, great. Successful SIGs can be a model to non-adopters if the strategies are helpful. That way buy-in happens over time.

I could see an instance in which this SIG might discover something that really should be addressed project-wide and adopted by all SIGs. In those cases, presenting a proposal or RFC to the TOC/SC for approval seems reasonable.

Thoughts @ezimanyi ?

**Comment by prophen** [comment]
We spent some time refining the messaging and scope of the proposed Contributor Experience SIG in response to your comments. 

Let us know if you approve 
@robzienert @ajordens @plumpy @ezimanyi  @ethanfrogers

File: cx-sig-charter-draft.md
Comment (ezimanyi) [reply cx-sig-charter-draft.md]: This model sounds great to me!  I really like the idea of this SIG helping other SIGs with best practices and tooling and agree (at least initially) that this would be best as opt-in. Just read through the text changes to the RFC which look good.

File: cx-sig-charter-draft.md
Comment (ezimanyi) [reply cx-sig-charter-draft.md]: Updates have provided clarify/resolved this.

**Review by ezimanyi — APPROVED** [review]
This looks great, thanks for all the effort in making this proposal and for the changes/responses to comments! 💯

**Review by robzienert — APPROVED** [review]
Excited to see what comes of this!

**Review by ethanfrogers — APPROVED** [review]
Can't wait to see the positive impact of this SIG!

**Comment by prophen** [comment]
@plumpy Are you able to approve? Do you have any feedback or concerns?

**Review by plumpy — APPROVED** [review]
I think this looks great and I look forward to seeing how this SIG can make the community better!

**Pull request merged**
